### PR TITLE
Mem buf redundant fields

### PR DESCRIPTION
**Pull request by liranoz12**

### CODE REVIEW

**Comment by liranoz12** [comment]
Notes:
1.  The second commit is only semantic change - i only remove the "path" from the struct and made the relevant changes
2.  Please verify that _serial_num_ filed of _mem_buff_desc_t_ is removed by #305 while merging this PR.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/834/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1097/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1112/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1134/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1142/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/841/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1164/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1169/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1187/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1203/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/850/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1204/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
@liranoz12 build fails under ./configure --enable-vmapoll
dev/cq_mgr.cpp: In member function 'int cq_mgr::vma_poll_reclaim_single_recv_buffer_helper(mem_buf_desc_t*)':
dev/cq_mgr.cpp:789: error: 'temp' was not declared in this scope
make[5]: *** [dev/cq_mgr.lo] Error 1
make[5]: *** Waiting for unfinished jobs....
dev/ring_simple.cpp: In member function 'void ring_simple::vma_poll_process_recv_buffer(mem_buf_desc_t*)':
dev/ring_simple.cpp:702: error: 'class mem_buf_desc_t' has no member named 'transport_header_len'
dev/ring_simple.cpp:926: error: 'class mem_buf_desc_t' has no member named 'transport_header_len'
dev/ring_simple.cpp: In member function 'bool ring_simple::rx_process_buffer(mem_buf_desc_t*, transport_type_t, void*)':
dev/ring_simple.cpp:986: error: 'class mem_buf_desc_t' has no member named 'transport_header_len'

File: src/vma/dev/cq_mgr.cpp [hunk 1]
```diff
@@ -785,15 +777,16 @@ int cq_mgr::vma_poll_reclaim_single_recv_buffer_helper(mem_buf_desc_t* buff)
...
+		buff->rx.tcp.gro = 0;
+		buff->rx.is_vma_thr = false;
+		buff->rx.vma_polled = false;
+		buff->rx.tcp.p_ip_h = NULL;
+		buff->rx.tcp.p_tcp_h = NULL;
+		buff->rx.udp.sw_timestamp.tv_nsec = 0;
+		buff->rx.udp.sw_timestamp.tv_sec = 0;
+		buff->rx.udp.hw_timestamp.tv_nsec = 0;
+		buff->rx.udp.hw_timestamp.tv_sec = 0;
+		temp->rx.hw_raw_timestamp = 0;
```
Comment (OphirMunk) [inline src/vma/dev/cq_mgr.cpp]: Suggesting chaning
```
temp->rx.hw_raw_timestamp = 0;
```
to
```
buff->rx.hw_raw_timestamp = 0;
```

File: src/vma/dev/ring_simple.cpp [hunk 2]
```diff
@@ -697,12 +697,12 @@ inline void ring_simple::vma_poll_process_recv_buffer(mem_buf_desc_t* p_rx_wc_bu
 		p_ip_h = (struct iphdr*)(p_rx_wc_buf_desc->p_buffer + transport_header_len);
 		ip_hdr_len = 20; //(int)(p_ip_h->ihl)*4;
 		struct tcphdr* p_tcp_h = (struct tcphdr*)((uint8_t*)p_ip_h + ip_hdr_len);
-		p_rx_wc_buf_desc->path.rx.p_ip_h        = p_ip_h;
-		p_rx_wc_buf_desc->path.rx.p_tcp_h       = p_tcp_h;
+		p_rx_wc_buf_desc->rx.tcp.p_ip_h        = p_ip_h;
+		p_rx_wc_buf_desc->rx.tcp.p_tcp_h       = p_tcp_h;
```
Comment (OphirMunk) [inline src/vma/dev/ring_simple.cpp]: Suggesting changing
```
p_rx_wc_buf_desc->transport_header_len  = transport_header_len;
```
to
```
p_rx_wc_buf_desc->rx.tcp.n_transport_header_len  = transport_header_len;
```

File: src/vma/dev/ring_simple.cpp [hunk 3]
```diff
@@ -906,25 +908,25 @@ inline void ring_simple::vma_poll_process_recv_buffer(mem_buf_desc_t* p_rx_wc_bu
...
-		p_rx_wc_buf_desc->path.rx.p_ip_h = p_ip_h;
-		p_rx_wc_buf_desc->path.rx.p_tcp_h = p_tcp_h;
+		p_rx_wc_buf_desc->rx.tcp.p_ip_h = p_ip_h;
+		p_rx_wc_buf_desc->rx.tcp.p_tcp_h = p_tcp_h;
 
 		// Find the relevant hash map and pass the packet to the rfs for dispatching
-		p_rfs = m_flow_tcp_map.get((flow_spec_tcp_key_t){p_rx_wc_buf_desc->path.rx.src.sin_addr.s_addr,
-			p_rx_wc_buf_desc->path.rx.dst.sin_port, p_rx_wc_buf_desc->path.rx.src.sin_port}, NULL);
+		p_rfs = m_flow_tcp_map.get((flow_spec_tcp_key_t){p_rx_wc_buf_desc->rx.src.sin_addr.s_addr,
+			p_rx_wc_buf_desc->rx.dst.sin_port, p_rx_wc_buf_desc->rx.src.sin_port}, NULL);
```
Comment (OphirMunk) [inline src/vma/dev/ring_simple.cpp]: Suggesting changing
```
p_rx_wc_buf_desc->transport_header_len  = transport_header_len;
```
to
```
p_rx_wc_buf_desc->rx.tcp.n_transport_header_len  = transport_header_len;
```

File: src/vma/dev/ring_simple.cpp [hunk 4]
```diff
@@ -979,8 +981,8 @@ bool ring_simple::rx_process_buffer(mem_buf_desc_t* p_rx_wc_buf_desc, transport_
 		p_ip_h = (struct iphdr*)(p_rx_wc_buf_desc->p_buffer + transport_header_len);
 		ip_hdr_len = 20; //(int)(p_ip_h->ihl)*4;
 		struct tcphdr* p_tcp_h = (struct tcphdr*)((uint8_t*)p_ip_h + ip_hdr_len);
-		p_rx_wc_buf_desc->path.rx.p_ip_h        = p_ip_h;
-		p_rx_wc_buf_desc->path.rx.p_tcp_h       = p_tcp_h;
+		p_rx_wc_buf_desc->rx.tcp.p_ip_h        = p_ip_h;
+		p_rx_wc_buf_desc->rx.tcp.p_tcp_h       = p_tcp_h;
```
Comment (OphirMunk) [inline src/vma/dev/ring_simple.cpp]: Suggesting changing
```
p_rx_wc_buf_desc->transport_header_len  = transport_header_len;
```
to
```
p_rx_wc_buf_desc->rx.tcp.n_transport_header_len  = transport_header_len;
```

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1212/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1215/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/858/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Merged #306

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1238/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1241/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1266/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1284/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1299/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1310/ for details (Mellanox internal link).

**Pull request closed without merging**